File: client/src/models/Offer.ts
import  Band  from './Band';
import Student  from './Student';
import  Recruiter  from './Recruiter';

export default interface Offer {
    bandName: string;
    offerId: number;
    amount: number;
    status: 'Pending' | 'Accepted' | 'Declined'; // Enum-like string literal
    offerDate: string;
    offerBandId: number; // Reference to the band making the offer
    studentId: string; // Reference to the student receiving the offer
    recruiterId: string; // Reference to the recruiter making the offer
    band?: Band;
    bandId: string;
    student?: Student;
    recruiter?: Recruiter;
    studentName?: string;
}

File: server/Models/Offer.cs
using System.ComponentModel.DataAnnotations.Schema;
using Models;

namespace Models
{
    public class Offer
    {
        public int OfferId { get; set; }
        
        public decimal Amount { get; set; }
        public OfferStatus Status { get; set; }  // Enum: Pending, Accepted, Rejected
        public DateTime OfferDate { get; set; }

        // Navigation Properties
        public int OfferBandId { get; set; }  // Foreign Key
        public Band Band { get; set; }
        public string StudentId { get; set; }  // Foreign Key
        public Student Student { get; set; }
        public string RecruiterId { get; set; }  // Foreign Key
        public Recruiter Recruiter { get; set; }
        public string? StudentName { get; set; }
        public string? RecruiterName { get;set; }
        public string? BandName { get; set; }
    }

    public enum OfferStatus
    {
        Pending,
        Accepted,
        Rejected
    }
}
